### PR TITLE
Added AND, EOR, ORA, BIT, CMP, CPX, CPY in CPU.cpp

### PR DESCRIPTION
**Pull request by arunabh-cloud**
Added AND, EOR, ORA, BIT, CMP, CPX, CPY in CPU.cpp

### CODE REVIEW

File: src/core/CPU.cpp [hunk 1]
```diff
@@ -961,7 +961,10 @@ Blaze::Cycles Blaze::CPU::executeADC(AddressingMode mode) {
 };
 
 Blaze::Cycles Blaze::CPU::executeAND(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address val = loadOperand(mode);
+	A &= val;
+	SetZeroNegFlags(A, true);
```
Comment (facekapow) [inline src/core/CPU.cpp:967]:
```suggestion
	setZeroNegFlags(A, true);
```

File: src/core/CPU.cpp [hunk 2]
```diff
@@ -971,22 +974,62 @@ Blaze::Cycles Blaze::CPU::executeASL(AddressingMode mode) {
 };
 
 Blaze::Cycles Blaze::CPU::executeBIT(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address val = loadOperand(mode);
+	setFlag(flags::z, !(A & val));
+	if (memoryAndAccumulatorAre8bit()) {
```
Comment (facekapow) [inline src/core/CPU.cpp:980]:
```suggestion
	if (memoryAndAccumulatorAre8Bit()) {
```

File: src/core/CPU.cpp [hunk 3]
```diff
@@ -971,22 +974,62 @@ Blaze::Cycles Blaze::CPU::executeASL(AddressingMode mode) {
...
 };
 
 Blaze::Cycles Blaze::CPU::executeCMP(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address val = loadOperand(mode);
+	Address temp = A - val;
+	setFlag(flags::z, (A == val));
+	setFlag(flags::c, (A >= val));
+	if (memoryAndAccumulatorAre8bit()) {
```
Comment (facekapow) [inline src/core/CPU.cpp:997]:
```suggestion
	if (memoryAndAccumulatorAre8Bit()) {
```

File: src/core/CPU.cpp [hunk 4]
```diff
@@ -971,22 +974,62 @@ Blaze::Cycles Blaze::CPU::executeASL(AddressingMode mode) {
...
 };
 
 Blaze::Cycles Blaze::CPU::executeCPX(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address val = loadOperand(mode);
+	Address temp = X - val;
+	setFlag(flags::z, (X == val));
+	setFlag(flags::c, (X >= val));
+	if (indexRegistersAre8bit()) {
```
Comment (facekapow) [inline src/core/CPU.cpp:1012]:
```suggestion
	if (indexRegistersAre8Bit()) {
```

File: src/core/CPU.cpp [hunk 5]
```diff
@@ -971,22 +974,62 @@ Blaze::Cycles Blaze::CPU::executeASL(AddressingMode mode) {
...
 };
 
 Blaze::Cycles Blaze::CPU::executeCPY(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address val = loadOperand(mode);
+	Address temp = Y - val;
+	setFlag(flags::z, (Y == val));
+	setFlag(flags::c, (Y >= val));
+	if (indexRegistersAre8bit()) {
```
Comment (facekapow) [inline src/core/CPU.cpp:1027]:
```suggestion
	if (indexRegistersAre8Bit()) {
```

File: src/core/CPU.cpp [hunk 6]
```diff
@@ -996,7 +1039,10 @@ Blaze::Cycles Blaze::CPU::executeDEC(AddressingMode mode) {
 };
 
 Blaze::Cycles Blaze::CPU::executeEOR(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address val = loadOperand(mode);
+	A ^= val;
+	SetZeroNegFlags(A, true);
```
Comment (facekapow) [inline src/core/CPU.cpp:1045]:
```suggestion
	setZeroNegFlags(A, true);
```

File: src/core/CPU.cpp [hunk 7]
```diff
@@ -1036,7 +1082,10 @@ Blaze::Cycles Blaze::CPU::executeLSR(AddressingMode mode) {
 };
 
 Blaze::Cycles Blaze::CPU::executeORA(AddressingMode mode) {
-	// TODO
+	// TODO TO CHECK
+	Address val = loadOperand(mode);
+	A |= val;
+	SetZeroNegFlags(A, true);
```
Comment (facekapow) [inline src/core/CPU.cpp:1088]:
```suggestion
	setZeroNegFlags(A, true);
```

**Review by facekapow — CHANGES_REQUESTED** [review]
LGTM; the only issue is you misspelled the method names (which is why the build is failing). Everything else looks good, though.

**Comment by arunabh-cloud** [comment]
Updated AND, ORA, EOR, BIT, CMP, CPX, CPY in CPU.cpp

**Pull request merged**